### DISCUSSION TITLE
Pack  direction bottom to top?

### DISCUSSION THREAD

**Discussion by l-o-u-i-s-g** [beeware/toga#2674] [Q&A] [answered]
Is there a "proper" way to add objects  on top of other objects  in a box?  

So far I can think of the following "hacks": 
 1 - doing a clear and re-adding in desired order. 
 2 - adding at bottom and swapping all, all the way down. I wonder what kind of visual effects that would do on a box with a lot of objects. 
 3 - create another box in parallel in the desired order than swap them. I guess this option would be more  UI efficient if the box as a lot of objects.

**Comment by mhsmith** [comment, marked as answer]
You can add a widget at any position you want with the [`insert`](https://toga.readthedocs.io/en/stable/reference/api/widgets/widget.html#toga.Widget.insert) method.

**Reply by l-o-u-i-s-g** [reply]
Thanks! 
That simplifies things a lot!  Is that documented somewhere?   I would have expected a list of methods within the documentation for box: 
https://toga.readthedocs.io/en/stable/reference/api/containers/box.html

**Reply by mhsmith** [reply]
It's in the base Widget class; the link is in my previous comment.
